src/main/kotlin/wordle/domain/Answer.kt

```diff
@@ -0,0 +1,58 @@
+package wordle.domain
+
+class Answer(private val answer: String) {
+
+    init {
+        require(answer.length == WORD_SIZE) { "[ERROR] 부적절한 글자 길이입니다." }
+        require(Words.contains(answer)) { "[ERROR] 목록에 존재하지 않는 단어입니다." }
+    }
+
+    fun compareToWord(word: String): MutableList<Mark> {
+        val result = MutableList(WORD_SIZE) { Mark.NONE }
+        val wordTable = createWordTable(word)
+        matchExact(word, result, wordTable)
+        matchExist(result, wordTable)
+        return result
+    }
+
+    private fun createWordTable(word: String): HashMap<Char, Int> {
+        val wordTable = HashMap<Char, Int>()
+        for (char in word) {
+            wordTable[char] = wordTable.getOrDefault(char, 0) + 1
+        }
+        return wordTable
+    }
+
+    private fun matchExact(word: String, result: MutableList<Mark>, wordTable: HashMap<Char, Int>) {
+        for (i in 0 until WORD_SIZE) {
```

Comment:
이렇게 map으로 처리하는 방법도 있겠네요! 꼭 이 방법이 더 좋다는건 아니고, 또 다른 방법도 존재한다는 내용의 코멘트입니다.
```suggestion
        (0 until WORD_SIZE).map { markExact(it, word, result, wordTable) }
```